src/mca/pcompress/zlib/compress_zlib.c

```diff
@@ -200,22 +202,26 @@ static bool zlib_decompress(uint8_t **outbytes, size_t *outlen, const uint8_t *i
 
 static bool decompress_string(char **outstring, uint8_t *inbytes, size_t len)
 {
-    int32_t len2;
+    uint32_t len2;
     bool rc;
     uint8_t *input;
 
     /* the first 4 bytes contains the uncompressed size */
     memcpy(&len2, inbytes, sizeof(uint32_t));
     /* add one to hold the NULL terminator */
-    ++len2;
+    if (++len2 == 0) {
+        /* set the default error answer */
+        *outstring = NULL;
+        return false;
+    }
 
     /* decompress the bytes */
     input = (uint8_t *) (inbytes + sizeof(uint32_t)); // step over the size
     rc = doit((uint8_t **) outstring, len2, input, len);
 
     if (rc) {
-        /* ensure this is NULL terminated! */
-        outstring[len2 - 1] = NULL;
+        /* ensure this is NUL terminated! */
```

Comment:
I believe `codespell` will bark about the spelling of `NUL` here - or at least, it certainly should! 😄

Reply:
I am referring to the ASCII "NUL" character, i.e. '\0'.

Reply:
👍